corehq/apps/accounting/utils.py

```diff
@@ -368,3 +368,20 @@ def cancel_future_subscriptions(domain_name, from_date, web_user):
             web_user=web_user,
             note="Cancelled due to changing subscription",
         )
+
+
+def is_downgrade(current_edition, next_edition):
```

Comment:
nit: An implementation with less complex branching could get the index of ```current_edition``` and ```next_edition``` in ```SoftwarePlanEdition.SELF_SERVICE_ORDER + [SoftwarePlanEdition.ENTERPRISE]``` and compare ```current_edition_index > next_edition_index```.